Finished Google Homepage

smeetthakkar opened the pull request:
This is the final version of the google homepage. Let me know if this is good to go.

smeetthakkar committed:
Updates README.md file

smeetthakkar committed:
Adds all project files

smeetthakkar committed:
Updates file paths

spwr007 commented on README.md:
```diff
@@ -0,0 +1,8 @@
+<p>The assigment was to recreate the Google Homepage. (the simple one with just a search box).</p>
+
+<p>Points to remember:</p>
+<ul>
```
https://guides.github.com/features/mastering-markdown/

smeetthakkar replied:
Noted. Thanks for sharing, I'll review and update. The new commit focuses on code related changes, will update the README.md in the next iteration.

smeetthakkar replied:
This is fixed as well. I updated the readme.md file.

spwr007 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" type="text/css" href="resources/style.css">
+    <title>Google</title>
```
Use 2 tab spaces

spwr007 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+<body>
+    <section class="center">
+        <center>
+            <img src="resources/images/logo.png" alt="company logo" class="company-logo">
```
best practise, void elemnts should be self closed, img, input, br, etc, etc

smeetthakkar replied:
Done.

spwr007 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+</head>
+<body>
+    <section class="center">
+        <center>
```
mostly avoid styling from tags, such as center, b, i, tag, dont use styling tags.
keep structure and styling separate.

smeetthakkar replied:
Noted. Thanks!

spwr007 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+            <div class="cta">
+                <center>
+                    <input type="submit" value="Google Search">
+                    <input type="submit" value="I'm Feeling Lucky">
```
input tags are form elements, should be used in form, without form no submit actions.
use button tags instead or input type button

smeetthakkar replied:
Implemented.

spwr007 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+            </div>
+        </center>
+    </section>
+    <footer>
+        <div class="fbar">
+            <div class="country">
+                <span>India</span>
+        </div>
+        <div class="fbar">
```
Indentation

spwr007 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+/* General Document Structure */
+
+* {
+    margin: 0;
```
2 tab spaces

smeetthakkar replied:
Updated css files.

spwr007 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+}
```
use reset css, recommended.

smeetthakkar replied:
My initial understanding was * {margin: 0, padding: 0, box-sizing: border-box; } resets CSS, however, I have researched a different approach, lmk if that meets your expectations?

vishalsg42 replied:
Have a look over this link: [reset css]( https://meyerweb.com/eric/tools/css/reset/)

spwr007 replied:
Reset CSS is not only for styling properties, but to maintain the styling along with units across browsers.
The above line does work but not perfectly for all tags. Follow reset css.

smeetthakkar replied:
Understood. Vishal shared a link, been using that as well.

spwr007 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+    height: 70px;
+}
+
+.cta input[type="submit"] {
```
i believe there are only two inputs ;p.

type= 'submit' not needed

smeetthakkar replied:
Haha, coolness ha! Thanks for the tip though!

spwr007 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+    color: rgba(0,0,0,.54);
+}
+
+.fbar {
```
keep one liner, when block contains only one css prop

smeetthakkar replied:
Implemented.

spwr007 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+
+.secondary-nav li {
+    margin-left: 30px;
+}
```
put an endline

smeetthakkar replied:
Implemented.

spwr007 reviewed (commented):
@smeetthakkar Reviewed

vishalsg42 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+        <div class="fbar">
+            <div class="footer-nav">
+                <ul class="primary-nav">
+                    <li>Advertising</li>
```
This should be clickeable. add anchor tag

smeetthakkar replied:
Implemented.

vishalsg42 commented on resources/style.css:
```diff
@@ -0,0 +1,179 @@
+/* General Document Structure */
+
```
add comments  for each section as you design

smeetthakkar replied:
Makes total sense, thanks, will keep that in mind in future projects.

vishalsg42 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
```
I understand you have committed all the files once. but try to start the project with git. Also, keep committing codes frequently as when you complete each section.

smeetthakkar replied:
Got it. And, you're right, I agree, having an initial commit on master and then resorting to branches is a good approach. Thanks for the tip!

vishalsg42 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+                    <span><ion-icon class="icon" name="search-outline"></ion-icon></span>
+                </div>
+                <div class="input-container">
+                    <input type="text" id="searchbox" name="searchbox" class="input-field">
```
```
<form>
For wrapping form element  use **form-group** classname
  <div class="form-group">
For input tags or form elems use **form-control** classname
       <input type="email" class="form-control" id="exampleInputEmail1" placeholder="Enter email">
  </div>
Submit button should not be wrapped. 
  <button type="submit" class="btn btn-primary">Submit</button>
</form>
```

smeetthakkar replied:
Sounds good. I've updated the approach but when I use forms in the future, I'll keep this in mind, thank you!

smeetthakkar committed:
Addresses Review Comments, Updates approach

1. Use 2 tab spaces for head tag
2. Closing void tags (img/logo)
3. Remove styling tags from markup
4. Changed approach for buttons, removed forms using anchor tags
5. Fixes any indentation issues
6. Uses reset command
7. One-liner for single style properties
8. CSS file has an endline
9. Nav has anchor tags
10. Comments for sections

smeetthakkar committed:
Addresses Review Comments, Updates approach

1. Use 2 tab spaces for head tag
2. Closing void tags (img/logo)
3. Remove styling tags from markup
4. Changed approach for buttons, removed forms using anchor tags
5. Fixes any indentation issues
6. Uses reset command
7. One-liner for single style properties
8. CSS file has an endline
9. Nav has anchor tags
10. Comments for sections

smeetthakkar commented:
@vishalsg42 @spwr007 Hi folks! Firstly, thanks a ton for the review!

I've addressed all your comments, please review and lmk your thoughts. FYI, I've update the approach. This means, the markup and styles were rewritten. I've included details in the commit message and the files for your perusal. Even added a tag on the previous commit just in case. 

@spwr007 I need to update the markdown for README.md, that's the only thing remaining. I'll take that up during the next iteration sometime this week. Will keep you posted. Thanks for sharing the resource.

smeetthakkar committed:
Merge branch 'stage' of github.com:smeetthakkar/google-hompage into stage

smeetthakkar reviewed (commented):
All changes reviewed/implemented, please check.

smeetthakkar committed:
Update README.md
